azurerm/provider.go

```diff
@@ -196,11 +207,27 @@ func providerConfigure(p *schema.Provider) schema.ConfigureFunc {
 			ClientSecret:              d.Get("client_secret").(string),
 			TenantID:                  d.Get("tenant_id").(string),
 			Environment:               d.Get("environment").(string),
+			UseMsi:                    d.Get("use_msi").(bool),
+			MsiEndpoint:               d.Get("msi_endpoint").(string),
 			SkipCredentialsValidation: d.Get("skip_credentials_validation").(bool),
 			SkipProviderRegistration:  d.Get("skip_provider_registration").(bool),
 		}
 
-		if config.ClientSecret != "" {
+		if config.UseMsi {
```

Comment:
Shall we use the same if-else logic as in the azurerm/config.go ?

Reply:
Do you mean from getAuthorizationToken ? I think the logic is slightly different in each

Reply:
Yes, I mean why we don't check if ClientSecret exists firstly.